### PR TITLE
[AIRFLOW-5343] Remove legacy way of pessimistic disconnect handling

### PR DESCRIPTION
**Pull request by Khrol**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5343
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

Based on discussions in https://github.com/apache/airflow/pull/5949 it was figured out that there is already pessimistic disconnect timeout handling. So instead of hand-written one only SQLAlchemy embedded way should be used.

'sqlalchemy~=1.3' is in `setup.py` requirements and `pool_pre_ping` appeared in SQLAlchemy 1.2.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

It's quite an edge case for integrated environment. I'm not quite sure how to test it.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -125,11 +125,7 @@ sql_alchemy_pool_recycle = 1800
 # Check connection at the start of each connection pool checkout.
 # Typically, this is a simple statement like “SELECT 1”.
 # More information here: https://docs.sqlalchemy.org/en/13/core/pooling.html#disconnect-handling-pessimistic
-sql_alchemy_pool_pre_ping = False
-
-# How many seconds to retry re-establishing a DB connection after
-# disconnects. Setting this to 0 disables retries.
-sql_alchemy_reconnect_timeout = 300
```
Comment (ashb) [inline airflow/config_templates/default_airflow.cfg:132]: What happens now with disconnects/errors?

Comment (Khrol) [reply airflow/config_templates/default_airflow.cfg:132]: If connection from the pool is not healthy, it's removed from the pool and new one is created (with three attempts). Other connections are invalidated from the pool as well. 

https://github.com/sqlalchemy/sqlalchemy/commit/f881dae8179b94f72ab0dc85d8f62be8c9ce2fe0#diff-31816cdb15e64b0af1b862f51abe1226R920 - the test visualizes this. 

This commit from SQLAlchemy is quite a good explanation.

Comment (bharathpalaksha) [reply airflow/config_templates/default_airflow.cfg:132]: pool_pre_ping argument should take care of it and the code for re connect is within sql alchemy library. We don't have to handle explicitly with new sql alchemy version

**Comment by Khrol** [comment]
Codecov report is confusing in terms of coverage. There are mainly code deletions here.

**Review by ashb — APPROVED** [review]

**Pull request merged**